lib/ffi/aspell/aspell.rb

```diff
@@ -13,8 +15,23 @@ module FFI
   # For more information see {FFI::Aspell::Speller}.
   #
   module Aspell
-    extend   FFI::Library
-    ffi_lib ['aspell', 'libaspell.so.15']
+    extend FFI::Library
+
+    # Allow custom pathing to library path
+    if !!ENV['ASPELL_LIB_PATH']
+      ffi_lib ENV['ASPELL_LIB_PATH']
+    else
+      case ::RbConfig::CONFIG['host_os']
+      when /linux/
+        ffi_lib ['aspell', 'libaspell.so.15']
+      when /darwin/
+        if File.exist?('/opt/boxen/homebrew/lib/libaspell.dylib')
```

Comment:
Boxen is just a framework where homebrew is moved to a different directory (`/opt/boxen` over `/usr/local`, so it's the same problem for both really, just a difference in pathing)

Linking is just the symlink to the lib directory from the "Cellar" where things are actually installed:

```
$ brew unlink aspell && brew link aspell
Unlinking /opt/boxen/homebrew/Cellar/aspell/0.60.6.1... 24 symlinks removed
Linking /opt/boxen/homebrew/Cellar/aspell/0.60.6.1... 24 symlinks created
$ ll /opt/boxen/homebrew/lib/libaspell.dylib
lrwxr-xr-x  1 petersouter  staff  45 Mar 12 20:41 /opt/boxen/homebrew/lib/libaspell.dylib -> ../Cellar/aspell/0.60.6.1/lib/libaspell.dylib
```

Unfortunately changing the file name but not specifying the brew path does not fix the issue. The following all give the same error:

```ruby
ffi_lib ['aspell', 'libaspell.so.15', 'libaspell']
```

```
/opt/boxen/rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ffi-1.9.18/lib/ffi/library.rb:147:in `block in ffi_lib': Could not open library 'aspell': dlopen(aspell, 5): image not found. (LoadError)
Could not open library 'libaspell.dylib': dlopen(libaspell.dylib, 5): image not found.
Could not open library 'libaspell.so.15': dlopen(libaspell.so.15, 5): image not found.
Could not open library 'libaspell.so.15.dylib': dlopen(libaspell.so.15.dylib, 5): image not found.
Could not open library 'libaspell': dlopen(libaspell, 5): image not found```

My path also has the lib directory in it:

```
echo $PATH | grep /opt/boxen/homebrew/lib/
/usr/local/heroku/bin:bin:/opt/boxen/packer:/opt/boxen/heroku/bin:/opt/boxen/rbenv/shims:/opt/boxen/rbenv/bin:/opt/boxen/ruby-build/bin:/opt/boxen/nodenv/bin:/opt/boxen/node-build/bin:/opt/boxen/homebrew/bin:/opt/boxen/homebrew/sbin:/opt/boxen/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/go/bin:/usr/local/MacGPG2/bin:/opt/puppetlabs/bin:/Users/petersouter/projects/bin:/opt/boxen/homebrew/lib/
```

So, as you say, this is some issue with homebrew. I've posted a thread on their forum about it to see if anyone has any ideas. For now I can simplify the PR to just add `libaspell` to the array of files to look for, as that's what it's called on OSX.